### ISSUE TITLE
Support Smudge Filters

### ISSUE THREAD

**Issue by MarkCallow** [CodeDocs/CodeDocs#10]
I use a git smudge filter to expand a $Date$ in some files. This needs to be run before the docs are generated. I have set up a script in the repo that installs the smudge filter in the git config file so what needs to be done after cloning the repo is the following:

``` bash
./install-gitconfig.sh
rm TODO.md include/ktx.h tools/toktx/toktx.cpp
git checkout TODO.md include/ktx.h tools/toktx/toktx.cpp
```

or if you are using Windows cmd prompt

``` cmd
install-gitconfig.bat
del TODO.md include/ktx.h tools/toktx/toktx.cpp
git checkout TODO.md include/ktx.h tools/toktx/toktx.cpp
```

Obviously this is repo specific so what is needed is a way to provide in .codedocs some commands to be run after the clone and before the documentation is generated. Something like

```
codedocs_before = 
```

**Comment by paulnovo** [comment]
Hmm, I have not used git smudge before, I'll look into it. I see that you are using HTML_TIMESTAMP, does this meet your requirements (although indirectly)?

**Comment by MarkCallow** [comment]
No, HTML_TIMESTAMP puts the time & date the docs are generated. The smudge filter I made expands $Date$ to the time and date the file was last modified.

The problem with the smudge/clean filter (any filter actually) is that is has to be set up in the git config file which is not something than can be stored in the repo. `install-gitconfig.sh` appends an `[include]` to the config file making it include the file `.gitconfig` from the repo. It only needs to be run when the repo is cloned but I guess CodeDocs is cloning every time it is notified of a commit.
